fault/verilator_target.py

```diff
@@ -1,5 +1,154 @@
 from .target import Target
-import magma.testing.verilator
+from magma.testing.function import testvectors
+import magma.config as config
+import inspect
+import os
+import subprocess
+from fault.array import Array
+import magma as m
+
+
+def harness(circuit, tests):
+
+    assert len(circuit.interface.ports.keys()) == len(tests[0])
+
+    test_vector_length = 0
+    for item in tests[0]:
+        if isinstance(item, Array):
+            test_vector_length += item.flattened_length
+        else:
+            test_vector_length += 1
+
+    source = '''\
```

Comment:
Eventually we should use a templating engine and move this into a separate template file.